### ISSUE TITLE
Minetest crashes after hitting F7 in GunshipPenguin's PvE server

### ISSUE THREAD

**Issue by Wuzzy2** [minetest/minetest#2279]
I just visited GunshipPenguin's PvE server and shortly after the world loaded, Minetest crashed. Before the crash, Minetest managed to render the interface, the HUD and the wieldhand. It appears that nothing of the world has been loaded so far.

The server's forum thread is here: https://forum.minetest.net/viewtopic.php?f=10&t=10616

Server hostname: minetest.gunshippenguin.com
Port: 30000

Last lines of debug.txt:

```
-------------
  Separator  
-------------

Irrlicht log: Irrlicht Engine version 1.8.1
Irrlicht log: Linux 3.17.6-1-ARCH #1 SMP PREEMPT Sun Dec 7 23:43:32 UTC 2014 x86_64
Irrlicht log: Using renderer: OpenGL 3.0
Irrlicht log: Mesa DRI Intel(R) Haswell Mobile : Intel Open Source Technology Center
Irrlicht log: OpenGL driver version is 1.2 or better.
Irrlicht log: GLSL version: 1.3
Irrlicht log: PNG warning: Interlace handling should be turned on when using png_read_image
Irrlicht log: Could not load texture: 
Irrlicht log: Could not load texture: 
Irrlicht log: Could not load texture: 
Irrlicht log: Could not load texture: 
Font size: 8 16

In thread 7f6bf2379780:
/build/minetest/src/minetest-0.4.11/src/game.cpp:3245: void Game::updateCamera(VolatileRunFlags*, irr::u32, irr::f32, float): Assertion 'playercao != __null' failed.
Debug stacks:
DEBUG STACK FOR THREAD 7f6bdffff700:
#0  virtual void* MeshUpdateThread::Thread()
DEBUG STACK FOR THREAD 7f6bf2379780:
#0  int main(int, char**)
(Leftover data: #1  void Client::step(float))
(Leftover data: #2  virtual void ClientEnvironment::step(float))
(Leftover data: #3  void Client::Receive())
(Leftover data: #4  void Client::ProcessData(irr::u8*, irr::u32, irr::u16))
(Leftover data: #5  void ItemStack::deSerialize(std::istream&, IItemDefManager*))
```

EDIT: I just visited the server again. Now I know more precisely when Minetest crashed: As I have hit the F7 key to switch the camera. This bug seems not to occour when the first parts of the worlds are actually displayed.

EDIT 2: I also know that I initially spawned in stone.

**Comment by nerzhul** [comment]
Confirmed.

**Comment by nerzhul** [comment]
Fixed in #2282 , commit https://github.com/minetest/minetest/commit/8252e1ecd097abd83a3e9ef95aba37a3d837e00b
